pyro/nn/module.py

```diff
@@ -97,8 +97,10 @@ def __getitem__(cls, Module):
             return PyroModule
         if Module in _PyroModuleMeta._pyro_mixin_cache:
             return _PyroModuleMeta._pyro_mixin_cache[Module]
+        bases = [PyroModule[b] for b in Module.__bases__
+                 if issubclass(b, torch.nn.Module)]
 
-        class result(Module, PyroModule):
+        class result(Module, *bases):
```

Comment:
This change seems clean and robust. It ensures distributivity of `PyroModule[-]`. For example if
```py
issubclass(GRU, RNNBase)
```
then
```py
issubclass(PyroModule[GRU], PyroModule[RNNBase])
```